example/simple_compute.cpp

```diff
@@ -0,0 +1,302 @@
+
+    init.disp.destroyShaderModule(shader_module, nullptr);
+    return 0;
+}
+
+int create_command_pool(Init& init, Data& data) {
+    VkCommandPoolCreateInfo pool_info = {};
+    pool_info.sType = VK_STRUCTURE_TYPE_COMMAND_POOL_CREATE_INFO;
+    pool_info.queueFamilyIndex = init.device.get_queue_index(vkb::QueueType::graphics).value();
+    init.disp.createCommandPool(&pool_info, nullptr, &data.command_pool);
+
+    VkCommandBufferAllocateInfo allocate_info = {};
+    allocate_info.sType = VK_STRUCTURE_TYPE_COMMAND_BUFFER_ALLOCATE_INFO;
+    allocate_info.commandPool = data.command_pool;
+    allocate_info.level = VK_COMMAND_BUFFER_LEVEL_PRIMARY;
+    allocate_info.commandBufferCount = 1;
+    init.disp.allocateCommandBuffers(&allocate_info, &data.command_buffer);
+
+    return 0;
+}
+
+uint32_t get_memory_index(Init& init, const uint32_t type_bits) {
+    uint32_t type_mask = type_bits;
+    VkPhysicalDeviceMemoryProperties mem_props = init.device.physical_device.memory_properties;
+    for (uint32_t i = 0; i < mem_props.memoryTypeCount; i++) {
+        if ((type_mask & 1) == 1) {
+            if ((mem_props.memoryTypes[i].propertyFlags &
+                    (VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT)) != 0) {
+                return i;
+            }
+        }
+        type_mask >>= 1;
+    }
+    return UINT32_MAX;
+}
+
+int create_buffers(Init& init, Data& data) {
+    VkBufferCreateInfo buffer_info = {};
+    buffer_info.sType = VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO;
+    buffer_info.size = sizeof(uint32_t);
+    buffer_info.usage = VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT | VK_BUFFER_USAGE_STORAGE_BUFFER_BIT;
+    init.disp.createBuffer(&buffer_info, nullptr, &data.buffer_a);
+
+    VkMemoryRequirements reqs;
+    init.disp.getBufferMemoryRequirements(data.buffer_a, &reqs);
+
+    VkMemoryAllocateInfo allocate_info = {};
+    allocate_info.sType = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO;
+    allocate_info.allocationSize = reqs.size;
+    allocate_info.memoryTypeIndex = get_memory_index(init, reqs.memoryTypeBits);
+    init.disp.allocateMemory(&allocate_info, nullptr, &data.memory_a);
+
+    init.disp.bindBufferMemory(data.buffer_a, data.memory_a, 0);
+
+    // Same for 2nd buffer
+    init.disp.createBuffer(&buffer_info, nullptr, &data.buffer_b);
+    init.disp.getBufferMemoryRequirements(data.buffer_b, &reqs);
+    allocate_info.allocationSize = reqs.size;
+    allocate_info.memoryTypeIndex = get_memory_index(init, reqs.memoryTypeBits);
+    init.disp.allocateMemory(&allocate_info, nullptr, &data.memory_b);
+    init.disp.bindBufferMemory(data.buffer_b, data.memory_b, 0);
+
+    // Update buffers with data
+    // (We are doing the world's most inefficient adding of 42 + 58 to get 100)
+    void* buffer_ptr;
+    init.disp.mapMemory(data.memory_a, 0, VK_WHOLE_SIZE, 0, &buffer_ptr);
```

Comment:
Kinda of the mind that the memory should be mapped when created and left mapped. No need to free them since its automatic and unnecessary un mapping (only to remap later) is a common perf loss for no gain in capabilities.